DIFM Lite: Revamp DIFM Lite flow

aneeshd16 opened the pull request:
### Changes proposed in this Pull Request

Context: pdh1Xd-iQ-p2

This PR will act as the parent PR and should be merged when the following PRs are completed and merged. Please do not commit directly to this branch.

- [x] New redux state for site info form (7171a59235e819377c428fb1fabe13b7de6f448c)
- [x] Site Information form step  #59986
- [x] Website content step  
    - [x] #60293
    - [x] #60436
    - [x] Step UI shell design Implementation     - #60548
    - [x] Calypso redux Data integration with step UI     - #60548
    - [x] Endpoint for website content persistence D74241-code
    - [x] Initialize the 4th page from the site category. https://github.com/Automattic/wp-calypso/pull/60998
    - [x] #61121
    - [x] Overall flow integration and completion https://github.com/Automattic/wp-calypso/pull/61029
- [x] Remove Typeform #60029 (blocked by #59986)
- [x] Redesigned New/Existing site and Site Picker steps #60479
- [x] Post-checkout screen
  - [x] User-action required https://github.com/Automattic/wp-calypso/pull/61070
  - [x] No user-action required: https://github.com/Automattic/wp-calypso/pull/60881

### Bug fixes
- [x] #61309
- [x] https://github.com/Automattic/wp-calypso/pull/60957
- [x] #61156

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

pdh1Xd-xv-p2
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->
#### Screenshots for translators

<img width="1890" alt="image" src="https://user-images.githubusercontent.com/5436027/154916540-1ec24574-57dc-47e7-a8cc-691036181110.png">
<img width="1906" alt="image" src="https://user-images.githubusercontent.com/5436027/154916597-73f71c68-ade5-45b1-9b3d-59dd2dc7c854.png">
<img width="1896" alt="image" src="https://user-images.githubusercontent.com/5436027/154916657-1344d400-0bae-49cb-9640-c5d11404606e.png">
<img width="1697" alt="image" src="https://user-images.githubusercontent.com/5436027/154916688-1984f028-f04e-4786-a309-46e53eee936b.png">
<img width="1568" alt="image" src="https://user-images.githubusercontent.com/5436027/154916716-f2cf4749-ad59-4715-800c-e115dc0ab560.png">
<img width="1391" alt="image" src="https://user-images.githubusercontent.com/5436027/154917328-a78acd1c-cad1-43c2-bbd0-61758c1dec4e.png">
<img width="1723" alt="image" src="https://user-images.githubusercontent.com/5436027/154917375-1f6c18e7-feb0-4305-9559-f3bb8494e3b3.png">

aneeshd16 committed:
Adds site info store to redux

aneeshd16 committed:
Adds site information step (#59986)

* Adds site information step

* Adds siteSlug to dependencies for setDesignOnSite

* Changes div to React fragment

* Email text fix

* Fixes unit test

* Rebased

* Removes hardcoded serial number

* Adds errors for form validation

* Fixes RTL bug

* Review feedback - 1

* Review Feedback - 2

* Created new component AccordionForm and related refactoring.
* Typo fix.

* Fix typo

* Fixes form styles

* Refactors function name

a8ci18n commented on client/signup/steps/site-info-collection/form-components.tsx:
```diff
@@ -0,0 +1,224 @@
+			<AddressField>
+				<TextAreaField
+					{ ...displayAddressProps }
+					label={ translate( 'Physical address' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Must be a physical address.' )` *ES Score: 7*

aneeshd16 committed:
Removes typeform dependency and config ids (#60029)

ddc22 committed:
The state and step scaffolding for the website-content step in DIFM onboarding (#60293)

* Initial state infrastructure for website content

* Move website content to different flow

* Fix signup destination call

* Refactor data structure

* Apply suggestions from code review

Co-authored-by: Aneesh Devasthale <aneeshd16@users.noreply.github.com>

Co-authored-by: Aneesh Devasthale <aneeshd16@users.noreply.github.com>

aneeshd16 committed:
Refactors AccordionForm into an independent component (#60334)

* Refactors AccordionForm into an independent component

* Rename useSections to sectionGenerator

a8ci18n commented on client/signup/accordion-form/form-components.tsx:
```diff
@@ -0,0 +1,224 @@
+			<AddressField>
+				<TextAreaField
+					{ ...displayAddressProps }
+					label={ translate( 'Physical address' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Must be a physical address.' )` *ES Score: 7*

ddc22 committed:
Component to upload image to media library and related UI (#60436)

* Adds site info store to redux

* Adds site information step (#59986)

* Adds site information step

* Adds siteSlug to dependencies for setDesignOnSite

* Changes div to React fragment

* Email text fix

* Fixes unit test

* Rebased

* Removes hardcoded serial number

* Adds errors for form validation

* Fixes RTL bug

* Review feedback - 1

* Review Feedback - 2

* Created new component AccordionForm and related refactoring.
* Typo fix.

* Fix typo

* Fixes form styles

* Refactors function name

* Removes typeform dependency and config ids (#60029)

* The state and step scaffolding for the website-content step in DIFM onboarding (#60293)

* Initial state infrastructure for website content

* Move website content to different flow

* Fix signup destination call

* Refactor data structure

* Apply suggestions from code review

Co-authored-by: Aneesh Devasthale <aneeshd16@users.noreply.github.com>

Co-authored-by: Aneesh Devasthale <aneeshd16@users.noreply.github.com>

* Refactors AccordionForm into an independent component (#60334)

* Refactors AccordionForm into an independent component

* Rename useSections to sectionGenerator

* A seperate component for picking files and uploading them to the wordpress user's relevant account.

* Apply suggestions from code review

Co-authored-by: Aneesh Devasthale <aneeshd16@users.noreply.github.com>

* Review fixes

* Update client/signup/steps/website-content/components.tsx

Co-authored-by: Aneesh Devasthale <aneeshd16@users.noreply.github.com>

* Cursor fix

Co-authored-by: Aneesh Devasthale <aneeshdevasthale@gmail.com>
Co-authored-by: Aneesh Devasthale <aneeshd16@users.noreply.github.com>

a8ci18n commented on client/signup/steps/website-content/wordpress-media-upload.tsx:
```diff
@@ -0,0 +1,143 @@
+									<img src={ uploadedImageUrl } alt={ imageCaption } />
+								</CroppedImage>
+							</Preview>
+							<Label>{ translate( 'Change File' ) }</Label>
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **133** times:
`translate( 'Change file' )` *ES Score: 8*

aneeshd16 committed:
Redesigns New/Existing Site screen (#60479)

a8ci18n commented on client/signup/steps/new-or-existing-site/index.tsx:
```diff
@@ -18,8 +21,10 @@ export default function NewOrExistingSiteStep( props: Props ): React.ReactNode {
 
-	const headerText = translate( 'Choose where you want us to build your site.' );
-	const subHeaderText = '';
+	const headerText = translate( 'Do it For Me' );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Do It For Me (Lite)' )` *ES Score: 7*

aneeshd16 committed:
Renames DIFM Lite option keys (#60549)

* Adds site information step (#59986)

* Adds site information step

* Adds siteSlug to dependencies for setDesignOnSite

* Changes div to React fragment

* Email text fix

* Fixes unit test

* Rebased

* Removes hardcoded serial number

* Adds errors for form validation

* Fixes RTL bug

* Review feedback - 1

* Review Feedback - 2

* Created new component AccordionForm and related refactoring.
* Typo fix.

* Fix typo

* Fixes form styles

* Refactors function name

* Renames option keys

* Deletes leftover files from rebase

ddc22 committed:
Website content UI shell and REDUX implementation (#60548)

* Adds site information step (#59986)

* Adds site information step

* Adds siteSlug to dependencies for setDesignOnSite

* Changes div to React fragment

* Email text fix

* Fixes unit test

* Rebased

* Removes hardcoded serial number

* Adds errors for form validation

* Fixes RTL bug

* Review feedback - 1

* Review Feedback - 2

* Created new component AccordionForm and related refactoring.
* Typo fix.

* Fix typo

* Fixes form styles

* Refactors function name

* Adds site information step (#59986)

* Adds site information step

* Adds siteSlug to dependencies for setDesignOnSite

* Changes div to React fragment

* Email text fix

* Fixes unit test

* Rebased

* Removes hardcoded serial number

* Adds errors for form validation

* Fixes RTL bug

* Review feedback - 1

* Review Feedback - 2

* Created new component AccordionForm and related refactoring.
* Typo fix.

* Fix typo

* Fixes form styles

* Refactors function name

* Website content overall UI

Fix for action types

Bug fix related to state saving

Decouple title from accordion

Write tests and fix logic issues

Cleanup title related remnants

* Review fixes

* Update client/signup/accordion-form/form-components.tsx

* Review fixes, schema validation tests, and schema fixes

* Move test state out of prod

* Update client/signup/accordion-form/form-components.tsx

Co-authored-by: Aneesh Devasthale <aneeshd16@users.noreply.github.com>

Co-authored-by: Aneesh Devasthale <aneeshd16@users.noreply.github.com>

aneeshd16 committed:
DIFM Lite: Initialise the 4th page from the site category (#60998)

* Adds site info store to redux

* The state and step scaffolding for the website-content step in DIFM onboarding (#60293)

* Initial state infrastructure for website content

* Move website content to different flow

* Fix signup destination call

* Refactor data structure

* Apply suggestions from code review

Co-authored-by: Aneesh Devasthale <aneeshd16@users.noreply.github.com>

Co-authored-by: Aneesh Devasthale <aneeshd16@users.noreply.github.com>

* Initialise the 4th page from the site category.

* Adds restaurant category

* Fixes unit test

* Fixes dependency for new users

Co-authored-by: Dehan <dehandecroos@gmail.com>

aneeshd16 committed:
DIFM Lite: Handler for submitting website content form (#61029)

* Adds site info store to redux

* The state and step scaffolding for the website-content step in DIFM onboarding (#60293)

* Initial state infrastructure for website content

* Move website content to different flow

* Fix signup destination call

* Refactor data structure

* Apply suggestions from code review

Co-authored-by: Aneesh Devasthale <aneeshd16@users.noreply.github.com>

Co-authored-by: Aneesh Devasthale <aneeshd16@users.noreply.github.com>

* Adds step action to submit Website Content form

* Adds missing import

Co-authored-by: Dehan <dehandecroos@gmail.com>

aneeshd16 committed:
DIFM Lite: Signup flow review fixes (#60957)

* Adds site info store to redux

* Adds site information step (#59986)

* Adds site information step

* Adds siteSlug to dependencies for setDesignOnSite

* Changes div to React fragment

* Email text fix

* Fixes unit test

* Rebased

* Removes hardcoded serial number

* Adds errors for form validation

* Fixes RTL bug

* Review feedback - 1

* Review Feedback - 2

* Created new component AccordionForm and related refactoring.
* Typo fix.

* Fix typo

* Fixes form styles

* Refactors function name

* The state and step scaffolding for the website-content step in DIFM onboarding (#60293)

* Initial state infrastructure for website content

* Move website content to different flow

* Fix signup destination call

* Refactor data structure

* Apply suggestions from code review

Co-authored-by: Aneesh Devasthale <aneeshd16@users.noreply.github.com>

Co-authored-by: Aneesh Devasthale <aneeshd16@users.noreply.github.com>

* Increases accordion clickable area

* Change description to tagline

* Adds address explainer text

* Make website content step responsive

* Adds space below the website content form

* Styling fixes

* Fixes image display

* Fix rebase errors

* Spacing fixes

* Removes redundant code

Co-authored-by: Dehan <dehandecroos@gmail.com>

aneeshd16 committed:
Adds an action required state to the DIFM Lite in progress screen (#61070)

* Adds an action required state to the DIFM Lite in progress screen

* Review fixes

* Replace div with fragment

aneeshd16 committed:
DIFM Lite: Misc. fixes (#61099)

* Adds custom processing screen text

* Fetch site after updating website content

* Display error message

* Adds additional step to the processing screen

aneeshd16 committed:
Copy and i18n fixes (#61195)

* Copy and i18n fixes

* Copy fix

ddc22 committed:
Limit website content filling page access (#61121)

* Redirect to relevent pages when DIFM site content  collection not applicable

* Depend on purchases

* Rely on sticker to decide if difm lite is purchased

* Bug fix site info not loaded

* Add user step to prevent logged out access

* Update client/signup/steps/website-content/index.tsx

Co-authored-by: Aneesh Devasthale <aneeshd16@users.noreply.github.com>

* Broken type fix

* Bug fix site information loading

* Prevent loading UI if relevent information has not yet loaded

Co-authored-by: Aneesh Devasthale <aneeshd16@users.noreply.github.com>

ddc22 committed:
Fix race condition issue (#61238)

aneeshd16 committed:
Minor copy fixes (#61246)

ddc22 committed:
Buggy homepage screen for DIFM (#61309)

* Buggy loading screen for DIFM

* Add is multiple sites loading request check

ddc22 reviewed (approved):
Confidence checks passed! LGTM! 🚢

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7149428

Hi @aneeshd16, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Website content not submitted`
- `Please provide the necessary information for the creation of your website. To access click on the button below.`
- `Add content to your website`
- `Please enter the following social media profile links if you have any.`
- `Physical address (if you want a map on your site)`
- `Saving your content`
- `Do it For Me`
- `Get your business up quickly with our experts building your responsive, professional website. Select your best option:`
- `%d. Name of your business`
- `Please enter the name of your business as you want it to appear on your new site.`
- `Please enter a valid site title.`
- `%d. Site Tagline`
- `A short tagline for your website.`
- `%d. Social Media Profiles`
- `Basic details about your website`
- `Website Content`
- `In this step, you will add your brand visuals, pages and media to be used on your website.`
- `Please provide the text you want to appear on your %(pageTitle)s page.`
- `Upload up to 3 images to be used on your %(pageTitle)s page.`
- `%(fieldNumber)d. %(pageTitle)s`
- `Please enter '%(pageTitle)s' content`
- `Image upload failed`

Thank you in advance!

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7149428).
